### PR TITLE
Move build from the 'dep' to 'mod' golang dependencies manager

### LINKED ISSUE #234: Move build from the 'dep' to 'mod' golang dependencies manager
It would be nice to move build from 'dep' to 'mod' golang dependencies manager.
@rigazilla what do you think about it. I can take care on migration, if community decide to do this.
Thanks in advance.

### PR DESCRIPTION
**Pull request by dmvolod**
Fixes #234

### CODE REVIEW

**Comment by dmvolod** [comment]
@rigazilla could you please to look at the PR. All tests are passed.

**Comment by rigazilla** [comment]
@dmvolod, yep I'll do asap. 
Maybe @slaskawi wants to have a look also.

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 language: go
 
 go:
-  - '1.11.x'
+  - '1.12.x'
```
Comment (slaskawi) [inline .travis.yml]: Perhaps 1.13?

File: .travis.yml [hunk 2]
```diff
@@ -19,8 +19,10 @@ jobs:
   include:
     - stage: "Unit Test"
       name: "Unit Test"
+      before_script:
+        - export GO111MODULE=on
       script:
-        - go get -u github.com/golang/dep/cmd/dep
+        - go mod tidy
```
Comment (slaskawi) [inline .travis.yml]: Perhaps this should be `go mod vendor`?

BTW, In Keycloak Operator we pushed the whole vendor directory. This made Travis builds blazing fast.

Comment (dmvolod) [reply .travis.yml]: Will change this for travis build, however the build time less than test several times

File: .travis.yml (hunk 1 above)
Comment (dmvolod) [reply .travis.yml]: Thanks, @slaskawi 
I will change this for travis build

File: .travis.yml (hunk 2 above)
Comment (galderz) [reply .travis.yml]: Our tests are not yet that fast (we need to implement mocking), so speed of adding vendor will probably be not so noticeable cos tests are the primary time consumer at the moment.

File: pkg/generated/clientset/versioned/clientset.go [hunk 3]
```diff
@@ -19,6 +19,8 @@ limitations under the License.
 package versioned
```
Comment (rigazilla) [inline pkg/generated/clientset/versioned/clientset.go]: Are all the changes in the `pkg/generated` folder related to the new deps manager?

Comment (dmvolod) [reply pkg/generated/clientset/versioned/clientset.go]: It's looks like related to changes in k8s.io/code-generator invocation and used version as in dep it was bind to kubernetes version not fixed (v0.17.0 // indirect) right now.

Comment (rigazilla) [reply pkg/generated/clientset/versioned/clientset.go]: uh ok, thanks!

File: build/boilerplate.go.txt [hunk 4]
```diff
@@ -0,0 +1,16 @@
+/*
```
Comment (galderz) [inline build/boilerplate.go.txt]: What's this file for?

Comment (dmvolod) [reply build/boilerplate.go.txt]: This is header for auto-generated files, as method moves from vendor/k8s.io/code-generator/generate-groups.sh invoke to the direct go run k8s.io/code-generator/cmd/client-gen call

**Comment by galderz** [comment]
Thx @dmvolod! Having this integrated should path the way for you to continue with the unit test work?

**Comment by dmvolod** [comment]
Yes, @galderz , sure. Thanks for merging this.

**Pull request merged**